src/Color.h

```diff
@@ -131,15 +131,15 @@ struct Color4ub {
 	Uint8 GetLuminance() const;
 	void Shade(float factor)
 	{
-		r *= static_cast<Uint8>(1.0f - factor);
-		g *= static_cast<Uint8>(1.0f - factor);
-		b *= static_cast<Uint8>(1.0f - factor);
+		r = static_cast<Uint8>(r * (1.0f - factor));
```

Comment:
I don't think this is mandatory, but I feel better if `r` is explicitly converted to float before its use.

Pros are that anyone is aware of conversions at a glance, cons are... none as the conversion must be
done anyway(?)

Reply:
You mean another static_cast<float>()? 😅

Reply:
>You mean another static_cast()?  :sweat_smile:

`r = static_cast<Uint8>(float(r) * (1.0f - factor));`

I know is "stylistically" uglier but implicit conversion still bite C/C++ programmers, and to me a better 'style' is a worse deal than a better 'grasp'...

I'm for merge anyway :smile:

Reply:
By writing ```r = static_cast<Uint8>(r * (1.0f - factor));``` then it's obvious that the inner ```r``` is casted to float in the multiplication, I think.

I'm more a C developer myself, but I know that in C++ static_cast<> is preferred. I'd rather not mix the two cast styles.

Reply:
IIRC a C-style cast would be `(float) r` not `float(r)` though I might be mistaken. The first reads as 'interpret r as a float', while the second reads 'convert r to float and use it in this expression', being an initializer instead of a cast.

Of course, the C++ spec is perfectly clear as to what's happening here; the byte gets implicitly converted to float as it's the larger datatype. I don't think we need to explicitly specify the cast.

Reply:
@Web-eWorks , @pcercuei : No needs, of course
> but I feel better if r is explicitly converted to float before its use.

And:
> I'm for merge anyway

thus, hurry up :+1: